services/blockchain-connector/shared/sdk/events.js

```diff
@@ -41,13 +48,46 @@ const events = [
 ];
 
 let eventsCounter;
+let lastBlockHeightEvent;
 
 const logError = (method, err) => {
 	logger.warn(`Invocation for ${method} failed with error: ${err.message}`);
 	logger.debug(err.stack);
 };
 
+const emitEngineEvents = async () => {
+	getNodeInfo().then(async nodeInfo => {
+		const { roundLength } = await getPosConstants();
+
+		setInterval(async () => {
+			const latestNodeInfo = await getNodeInfo(true);
+			const { syncing, height, genesisHeight } = latestNodeInfo;
+			const isNodeSyncComplete = !syncing;
+
+			if (isNodeSyncComplete) {
+				if (!lastBlockHeightEvent || height > lastBlockHeightEvent) {
+					lastBlockHeightEvent = height;
+					const newBlock = await getBlockByHeight(height);
+					Signals.get(EVENT_CHAIN_BLOCK_NEW).dispatch({ blockHeader: newBlock.header });
+
+					if ((height - genesisHeight) % roundLength === 1) {
+						const bftParameters = await getBFTParameters(height);
+						Signals.get(EVENT_CHAIN_VALIDATORS_CHANGE).dispatch({
+							nextValidators: bftParameters.validators,
+							certificateThreshold: bftParameters.certificateThreshold,
+							precommitThreshold: bftParameters.precommitThreshold,
+						});
+					}
+				}
+			}
+		}, Math.min(3, nodeInfo.genesis.blockTime) * 1000);
```

Comment:
To handle error scenario.
```suggestion
		const { roundLength } = await getPosConstants();
		const blockTime = nodeInfo?.genesis?.blockTime || 10;

		setInterval(async () => {
			const latestNodeInfo = await getNodeInfo(true);
			const { syncing, height, genesisHeight } = latestNodeInfo;
			const isNodeSyncComplete = !syncing;

			if (isNodeSyncComplete) {
				if (!lastBlockHeightEvent || height > lastBlockHeightEvent) {
					lastBlockHeightEvent = height;
					const newBlock = await getBlockByHeight(height);
					Signals.get(EVENT_CHAIN_BLOCK_NEW).dispatch({ blockHeader: newBlock.header });

					if ((height - genesisHeight) % roundLength === 1) {
						const bftParameters = await getBFTParameters(height);
						Signals.get(EVENT_CHAIN_VALIDATORS_CHANGE).dispatch({
							nextValidators: bftParameters.validators,
							certificateThreshold: bftParameters.certificateThreshold,
							precommitThreshold: bftParameters.precommitThreshold,
						});
					}
				}
			}
		}, Math.min(3, blockTime) * 1000);
```